EventListener/InvalidationSubscriber.php

```diff
@@ -220,11 +226,60 @@ private function invalidateRoutes(array $routes, Request $request)
                 }
             }
 
+            if (null === $route->getParams()) {
+                $typeInvalidation = "BAN";
+            }
+
+            $typeInvalidation = $route->type ? $route->type : $typeInvalidation;
+
+            if ($typeInvalidation === "BAN") {
+                $this->cacheManager->invalidateRegex($this->generateRegexPath($route->getName(), $params));
+
+                continue;
+            }
+
             $this->cacheManager->invalidateRoute($route->getName(), $params);
         }
     }
 
     /**
+     * Set all route params as widlcard by default
+     *
+     * @param string $routeName
+     *
+     * @return array
+     */
+    protected function preFillParams($routeName)
+    {
+        $params = array();
+        $route = $this->router->getRouteCollection()->get($routeName->getName());
+        $pathVariables = $route->compile()->getPathVariables();
+        foreach ($pathVariables as &$param) {
+            $params[$param] = '(.*)';
+        }
+
+        return $params;
+    }
+
+    /**
+     * Generate url regex by route path
+     *
+     * @param string $routeName
+     * @param array $params
+     *
+     * @return string
+     */
+    protected function generateRegexPath($routeName, $params)
```

Comment:
you should use private visibility, not protected one (protected means it is an extension point covered by BC concerns, which must be avoided whenever we can to simplify the maintenance).

And array arguments should be typehinted